File: demo/module4/src/main/java/com/jst/module4/Module4Component.java
package com.jst.module4;

import android.content.Context;
import android.util.Log;

import com.jst.compinit.IComponent;
import com.jst.compinit.annotation.Component;

@Component
public class Module4Component implements IComponent {
    @Override
    public void init(Context context) {
        Log.i("compinit_test","Module4Component init");
    }
}

File: demo/module4/src/main/java/com/jst/module4/Module4Component.kt
package com.jst.module4

import android.content.Context
import android.util.Log
import com.jst.compinit.IComponent
import com.jst.compinit.annotation.Component

@Component
class Module4Component : IComponent {
    override fun init(context: Context) {
        Log.i("compinit_test", "Module4Component init")
    }
}